src/main/main.ts

```diff
@@ -33,37 +46,221 @@ export default class AppUpdater {
 
 let mainWindow: BrowserWindow | null = null;
 
-ipcMain.on('peace', async (event, arg) => {
-  const channel: string = arg[0];
-  const peaceInstalled = await registry.isPeaceInstalled();
-  if (!peaceInstalled) {
-    const reply: TError = { errorCode: ErrorCode.PEACE_NOT_INSTALLED };
-    event.reply(channel, reply);
-    return;
+/** ----- Equalizer APO Implementation ----- */
+
+// Load initial state from local state file
+const state: IState = fetch();
+let configPath = '';
+
+const retryHelper = async (attempts: number, f: () => unknown) => {
+  for (let i = 0; i < attempts; i += 1) {
+    try {
+      await f();
+      return;
+    } catch (e) {
+      if (i === attempts) {
+        throw new Error(`Failed to perform action after ${attempts} retrires`);
+      }
+      // TODO: maybe add a backoff here?
+      await new Promise((resolve) => {
+        setTimeout(resolve, 500);
+      });
+    }
   }
+};
 
-  const peaceHWnd = getPeaceWindowHandle();
-  const foundPeace = isPeaceRunning(peaceHWnd);
+const handleError = (
+  event: Electron.IpcMainEvent,
+  channel: ChannelEnum | string,
+  errorCode: ErrorCode
+) => {
+  const reply: TError = { errorCode };
+  event.reply(channel, reply);
+};
 
-  if (!foundPeace) {
-    const reply: TError = { errorCode: ErrorCode.PEACE_NOT_RUNNING };
-    event.reply(channel, reply);
+const handleUpdate = async (
+  event: Electron.IpcMainEvent,
+  channel: ChannelEnum | string,
+  checkConfig = false
```

Comment:
actually we should be extracting the checkConfig aspect of this function

Reply:
The type on `checkConfig` gets auto-removed by the linter because of this recommendation:

> Type boolean trivially inferred from a boolean literal, remove type annotation.eslint [@typescript-eslint/no-inferrable-types](https://typescript-eslint.io/rules/no-inferrable-types)